service-base/src/test/java/org/eclipse/hono/service/credentials/CredentialsPlainPasswordHelperTest.java

```diff
@@ -0,0 +1,179 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.credentials;
+
+import io.vertx.core.json.JsonObject;
+import org.eclipse.hono.util.CredentialsConstants;
+import org.junit.Test;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static org.eclipse.hono.service.credentials.CredentialsPlainPasswordHelper.hashPwdAndUpdateSecret;
+import static org.eclipse.hono.service.credentials.CredentialsPlainPasswordHelper.hashWithSha;
+import static org.junit.Assert.*;
+
+/**
+ * Unit tests for {@link CredentialsPlainPasswordHelper}.
+ */
+public class CredentialsPlainPasswordHelperTest {
+
+    private static final String PASSWORD = "mylittlesecret";
+    private static final byte[] SALT = "saltnpepper".getBytes(StandardCharsets.UTF_8);
+
+    /**
+     * Tests that password is correctly hashed with sha-256.
```

Comment:
FMPOV the description of the test methods should start with `Verifies that`, implying that we expect the test to turn out positively, right?